### ISSUE TITLE
add parameter workdir

### ISSUE THREAD

**Issue by milahu** [webfp/tor-browser-selenium#209]
currently tbselenium does

```py
chdir(self.tbb_browser_dir)
```

which requires a writable `tbb_browser_dir`

but this should also work with a read-only system-wide installation of torbrowser
and it should be possible to use a workdir in tmpfs

**Comment by gunesacar** [comment]
Thanks for filing the issue. This is a busy time of the year, so it'll be a while until I get to this. 

> sudo chown 0:0 "tbb_path/Browser/.tor project/"

A clarifying question: I am not familiar with the `.tor project/` directory. Could you link to the source code you are referring to or some public documentation?

> but my actual issue is...

Just to clarify, you get this issue when you use `tbselenium` on nixos?

**Comment by milahu** [comment]
yes, this is on nixos linux, see my [tbselenium.nix](https://github.com/milahu/nur-packages/raw/master/pkgs/python3/pkgs/tbselenium/default.nix)

selenium seems to use `$tbb_browser_dir` as `$HOME` hence `$tbb_browser_dir/.tor project/`


on linux, the torbrowser firefox profile is stored in `~/.tor project/firefox/xxxxxxxx.default/`
for firefox that is `~/.mozilla/firefox/xxxxxxxx.default/`
for librewolf that is `~/.librewolf/xxxxxxxx.default/`
etc

this is not really documented...

i have not found the code for
`lower($MOZ_APP_VENDOR/$MOZ_APP_BASENAME)`


https://gitlab.torproject.org/tpo/applications/tor-browser/-/blob/tor-browser-115.12.0esr-13.5-1/browser/confvars.sh

```sh
MOZ_APP_VENDOR="Tor Project"
```

https://searchfox.org/mozilla-central/source/build/moz.configure/init.configure

```py
def moz_app_basename(value, target_is_android):
    if value:
        return value[0]
    if target_is_android:
        return "Fennec"
    return "Firefox"
```

https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/20497#note_2873088

> The profile for torbrowser ends up in `$HOME/.tor project/torbrowser`

https://issues.guix.gnu.org/42380#42-lineno202

```guile
;; Profile location (relative to "~/.").  Default is
;; lower($MOZ_APP_VENDOR/$MOZ_APP_BASENAME), which is:
;; ~/.tor project/firefox.
(setenv "MOZ_APP_PROFILE" "torbrowser/browser")
```
